lib/cookie/cookieJar.ts

```diff
@@ -330,12 +327,22 @@ export class CookieJar {
     }
     let syncErr: Error | null = null
     let syncResult: T | undefined = undefined
-    fn.call(this, (error: Error | null, result?: T | undefined) => {
-      syncErr = error
-      syncResult = result
-    })
+
+    try {
+      fn.call(this, (error: Error | null, result?: T | undefined) => {
+        syncErr = error
+        syncResult = result
+      })
+    } catch (err) {
+      if (err instanceof Error) {
+        syncErr = err
+      } else {
+        throw err
+      }
+    }
+
     // These seem to be false positives; it can't detect that the value may be changed in the callback
```

Comment:
```suggestion
```
Comment is for the removed `// eslint-disable`

Reply:
Fixed